xbmc/interfaces/json-rpc/AddonsOperations.cpp

```diff
@@ -79,8 +79,18 @@ JSONRPC_STATUS CAddonsOperations::GetAddons(const std::string &method, ITranspor
     {
       if (!enabled.isBoolean())
       {
-        CAddonMgr::Get().GetAllAddons(typeAddons, false);
-        CAddonMgr::Get().GetAllAddons(typeAddons, true);
+        for (int i = ADDON_UNKNOWN+1; i < ADDON_MAX; ++i)
+        {
+          if (ADDON_REPOSITORY != (TYPE)i)
+          {
+            VECADDONS diabled;
+            if (CAddonMgr::Get().GetAddons((TYPE)i, diabled, false))
+              addons.insert(addons.end(), diabled.begin(), diabled.end());
+            VECADDONS enabled;
+            if (CAddonMgr::Get().GetAddons((TYPE)i, enabled, true))
+              addons.insert(addons.end(), enabled.begin(), enabled.end());
+          }
+        }
```

Comment:
@Montellese I'm not sure if to do with the json-rpc function so I've moved the old code here. IMO returning duplicates here seems weird, but what do you think? Keep or remove?